cmd/synthetic-monitoring-agent/http.go

```diff
@@ -48,9 +50,12 @@ func NewMux(opts MuxOpts) *Mux {
 	isReady.Store(false)
 	router.Handle("/ready", opts.isReady)
 
-	// disconnect this agent from the API
-	if opts.disconnectEnabled {
-		router.Handle("/disconnect", http.HandlerFunc(disconnectHandler))
+	// disconnect this agent from the API if disconnect is enabled with a password
+	if opts.disconnectSecret != "" {
+		disconnectHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			disconnectHandler(w, r, opts.disconnectSecret)
+		})
+		router.Handle("/disconnect", disconnectHandler)
```

Comment:
I don't like that an API is exposed / not-exposed based on command line flags. I think the endpoint should always be present, it should return different values based on whether it's enabled or not (e.g. Forbidden vs Unauthorized)